src/components/Settings/Language/Language.container.js

```diff
@@ -87,10 +83,34 @@ export class LanguageContainer extends Component {
       avaliableAndDownloadablesLangs: [],
       downloadablesOnly: []
     },
-    downloadLangLoading: true,
+    downloadLangLoading: false,
     downloadingLangError: { ttsError: false, langError: false }
   };
 
+  componentDidMount = async () => {
+    const { isdownloading } = this.props.downloadingLang;
+    this.setState({
+      downloadablesLangs: isAndroid()
+        ? this.prepareDownloadablesLenguages()
+        : {
+            avaliableAndDownloadablesLangs: [],
+            downloadablesOnly: []
+          }
+    });
+    if (isdownloading) {
+      this.setState({ downloadLangLoading: true });
+    } else {
+      this.setState({ downloadLangLoading: false });
+    }
+  };
+
+  componentDidUpdate = async () => {
+    const { isdownloading, isUpdated } = this.props.downloadingLang;
+    if (isdownloading && isUpdated) {
```

Comment:
this is intentional in order to allow asynchronous behavior between `speechProvider` and this component. An infinite loop should not happen as we are removing the prop once it was used.

Reply:
Im not sure if we are removing the prop before a componentDidUpdate is fired again. componentDidUpdate() is fired after every change. Like a setState(). one of the first lines on lookDownloadingLang() is `this.setState({ downloadLangLoading: false });`  and this for example is firing componentDidUpdate again. 
see src/components/Settings/Language/Language.container.js:435 and https://es.reactjs.org/docs/react-component.html#componentdidupdate
now we are getting this error: `Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.`
I think that the change recommended is not breaking the asynchronous behavior between the Speech Provider and this component because lookDownloadingLang() is fired inmediatly after voices and langs are updated.

Reply:
Like Mostly times. You are on truth. On the new PR #1229 I refactor the function to set isUpdated: false immediately after using it and the loop is not executed.